docs: manpage, keycodes, json, editorconfig

justinmk opened the pull request:
- [x] json formatting https://github.com/neovim/neovim/discussions/35683

justinmk committed:
docs: manpage, keycodes, json

zeertzjq commented on runtime/doc/intro.txt:
```diff
@@ -237,73 +237,74 @@ These names for keys are used in the documentation.  They can also be used
-<M-…>           alt-key or meta-key             *META* *ALT* *<M-*
-<A-…>           same as <M-…>                   *<A-*
-<T-…>           meta-key when it's not alt      *<T-*
-<D-…>           command-key or "super" key      *<D-*
+<C-…>           Control-key                     *control* *ctrl* *<C-*
+<M-…>           Alt-key or meta-key             *META* *ALT* *<M-*
+<A-…>           Same as <M-…>                   *<A-*
+<T-…>           Meta-key, when it's not alt     *<T-*
+<D-…>           Command-key or "super" key      *<D-*
```
The capitalization here is inconsistent.

justinmk replied:
I missed the `shift-key` 🤦

justinmk committed:
docs: move editorconfig.txt into plugins.txt

It helps to have plugins living in one common area, because it signals
to users the mechanisms for controlling them, which are typically driven
by keymaps and autocmds rather than builtin options.

We can always revisit if plugins.txt gets "too big" (for example, we may
want to introduce "project.txt" for the project concept, where
editorconfig and 'exrc' are relevant), but for now it's rather unusual
for editorconfig.txt to have its own dedicated helpfile.

justinmk commented on runtime/doc/intro.txt:
```diff
@@ -237,73 +237,74 @@ These names for keys are used in the documentation.  They can also be used
+<EOL>           End-of-line (can be <CR>, <NL> or <CR><NL>,
+                Depends on system and 'fileformat')     *<EOL>*
+<Ignore>        Cancel wait-for-character               *<Ignore>*
+<NOP>           Do nothing (no-op). Useful in mappings. *<Nop>*
+                <Ignore> is a key, <NOP> is "absence of a key".
```
Added note from @zeertzjq

justinmk commented on runtime/doc/plugins.txt:
```diff
@@ -37,6 +37,91 @@ Help-link		Loaded	Short description
 |tohtml|		Yes	Convert buffer to html, syntax included
 
+==============================================================================
+Builtin plugin: editorconfig                                    *editorconfig*
```
Moved editorconfig to plugins.txt.

We can always revisit if plugins.txt gets "too big", but for now it's rather unusual for editorconfig.txt to have its own dedicated helpfile.

It helps to have plugins living in one common area, because it signals to users the mechanisms for controlling them, which are typically driven by keymaps and autocmds rather than builtin options.

zeertzjq commented on runtime/doc/intro.txt:
```diff
@@ -237,73 +237,74 @@ These names for keys are used in the documentation.  They can also be used
+<kEqual>        Keypad =                        *keypad-equal*
+<kEnter>        Keypad Enter                    *keypad-enter*
+<k0> - <k9>     Keypad 0 to 9                   *keypad-0* *keypad-9*
 <S-…>           shift-key                       *shift* *<S-*
```
```suggestion
<S-…>           Shift-key                       *shift* *<S-*
```
